rs/tests/networking/canister_http_correctness_test.rs

```diff
@@ -610,30 +814,1006 @@ pub fn test_http_calls_to_ic_fails(env: TestEnv) {
+    //       content-type: application/octet-stream
+    //       content-length: 0
+    //       access-control-allow-origin: *
+    //       access-control-allow-credentials: true
+    //   }
+
+    let header_size = 142;
+    let max_response_bytes = Some(header_size + n - 1);
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: CanisterHttpRequestArgs {
+                url,
+                headers: BoundedHttpHeaders::new(vec![]),
+                method: HttpMethod::GET,
+                body: None,
+                transform: None,
+                max_response_bytes,
+            },
+            cycles: 500_000_000_000,
+        },
+    ));
+
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::SysFatal,
+            ..
+        })
+    );
+}
+
+fn test_non_ascii_url_is_rejected(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+    let expected_body = "안녕하세요";
+
+    let url = format!(
+        "https://[{}]:20443/{}/{}",
+        webserver_ipv6, "ascii", expected_body
+    );
+
+    let max_response_bytes = 666;
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers: BoundedHttpHeaders::new(vec![]),
+        method: HttpMethod::GET,
+        body: Some("".as_bytes().to_vec()),
+        transform: None,
+        max_response_bytes: Some(max_response_bytes),
+    };
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: request.clone(),
+            cycles: 500_000_000_000,
+        },
+    ));
+
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::SysFatal,
+            ..
+        })
+    );
+}
+
+/// Test that the response body returned is the same as the requested path.
+fn test_max_url_length(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+
+    let base_url = format!("https://[{}]:20443/{}/", webserver_ipv6, "ascii");
+    let remaining_space = MAX_CANISTER_HTTP_URL_SIZE - base_url.len();
+    let expected_body = "x".repeat(remaining_space);
+
+    let url = format!("{}{}", base_url, expected_body);
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers: BoundedHttpHeaders::new(vec![]),
+        method: HttpMethod::GET,
+        body: Some("".as_bytes().to_vec()),
+        transform: None,
+        max_response_bytes: None,
+    };
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: request.clone(),
+            cycles: 500_000_000_000,
+        },
+    ))
+    .expect("Request is successful.");
+
+    assert_matches!(&response, RemoteHttpResponse {body, status: 200, ..} if *body == expected_body);
+    assert_http_response(&response);
```

Comment:
let's also check that the url is not trimmed: the body should be equal to `expected_body`